### PR TITLE
Nessie CLI: publish Docker images

### PR DESCRIPTION
**Pull request by adutra**

### CODE REVIEW

File: site/in-dev/cli.md [hunk 1]
```diff
@@ -5,35 +5,55 @@ title: "Nessie CLI"
...
+=== "Docker image"
+    Nessie CLI has a Docker image hosted on [GitHub Container Registry]. Images are also mirrored to [Quay.io].
 
-```bash
-java -jar nessie-cli-<version>.jar
-```
+    Running the docker image can be done as follows:
+    
+    ```bash
+    docker run --rm -it ghcr.io/projectnessie/nessie-cli:::NESSIE_VERSION::
```
Comment (dimas-b) [inline site/in-dev/cli.md:37]: nit: maybe use `latest` for "in-dev" docs?

**Review by dimas-b — APPROVED** [review]

File: site/in-dev/cli.md (hunk 1 above)
Comment (adutra) [reply site/in-dev/cli.md:37]: But isn't this page also going to serve for released versions? Is there a way to input `latest` for nightly and a version number elsewhere?

Comment (adutra) [reply site/in-dev/cli.md:37]: Took a stab at it: #8939

Comment (snazy) [reply site/in-dev/cli.md:37]: > nit: maybe use `latest` for "in-dev" docs?

`in-dev` serves the current "development" docs _and_ becomes the released version docs.

File: site/docs/downloads/index.md [hunk 2]
```diff
@@ -27,13 +27,13 @@ The main Nessie server serves the Nessie repository using the Iceberg REST API a
 Docker images are multiplatform images for amd64, arm64, ppc64le, s390x.
 
 === "GitHub Container Registry"
-    [GitHub Container Registry](https://ghcr.io/projectnessie/nessie)
+    [![ghcr.io GitHub Container Registry](https://img.shields.io/maven-central/v/org.projectnessie.nessie/nessie?label=ghcr.io+Docker&logo=docker&color=3f6ec6&style=for-the-badge&logoColor=white)](https://github.com/projectnessie/nessie/pkgs/container/nessie)
```
Comment (snazy) [inline site/docs/downloads/index.md:30]: Better remove the badge for consistency

File: site/docs/downloads/index.md [hunk 3]
```diff
@@ -27,13 +27,13 @@ The main Nessie server serves the Nessie repository using the Iceberg REST API a
...
     ```bash
     docker pull ghcr.io/projectnessie/nessie:{{ versions.nessie }}
     docker run -p 19120:19120 -p 9000:9000 ghcr.io/projectnessie/nessie:{{ versions.nessie }}
     ```
 === "Quay.io"
-    [quay.io](https://quay.io/repository/projectnessie/nessie?tab=tags)
+    [![quay.io Quay](https://img.shields.io/maven-central/v/org.projectnessie.nessie/nessie?label=quay.io+Docker&logo=docker&color=3f6ec6&style=for-the-badge&logoColor=white)](https://quay.io/repository/projectnessie/nessie?tab=tags)
```
Comment (snazy) [inline site/docs/downloads/index.md:36]: Better remove the badge for consistency

File: site/docs/downloads/index.md [hunk 4]
```diff
@@ -62,6 +62,23 @@ java -jar nessie-quarkus-{{ versions.nessie }}-runner.jar
 
 [Nessie CLI](/nessie-latest/cli/) is both a command-line interface but primarily a REPL.
 
+### Docker image
+
+Docker images are multiplatform images for amd64, arm64, ppc64le, s390x.
+
+=== "GitHub Container Registry"
+    [![ghcr.io GitHub Container Registry](https://img.shields.io/maven-central/v/org.projectnessie.nessie/nessie?label=ghcr.io+Docker&logo=docker&color=3f6ec6&style=for-the-badge&logoColor=white)](https://github.com/projectnessie/nessie/pkgs/container/nessie-cli)
```
Comment (snazy) [inline site/docs/downloads/index.md:70]: Better remove the badge for consistency

File: site/docs/downloads/index.md [hunk 5]
```diff
@@ -62,6 +62,23 @@ java -jar nessie-quarkus-{{ versions.nessie }}-runner.jar
...
+    ```bash
+    docker pull ghcr.io/projectnessie/nessie-cli:{{ versions.nessie }}
+    docker run -it ghcr.io/projectnessie/nessie-cli:{{ versions.nessie }}
+    ```
+=== "Quay.io"
+    [![quay.io Quay](https://img.shields.io/maven-central/v/org.projectnessie.nessie/nessie?label=quay.io+Docker&logo=docker&color=3f6ec6&style=for-the-badge&logoColor=white)](https://quay.io/repository/projectnessie/nessie-cli?tab=tags)
```
Comment (snazy) [inline site/docs/downloads/index.md:76]: Better remove the badge for consistency

File: site/docs/downloads/index.md [hunk 6]
```diff
@@ -81,13 +98,13 @@ java -jar nessie-cli-{{ versions.nessie }}.jar
 Docker images are multiplatform images for amd64, arm64, ppc64le, s390x.
 
 === "GitHub Container Registry"
-    [GitHub Container Registry](https://github.com/projectnessie/nessie/pkgs/container/nessie-gc)
+    [![ghcr.io GitHub Container Registry](https://img.shields.io/maven-central/v/org.projectnessie.nessie/nessie?label=ghcr.io+Docker&logo=docker&color=3f6ec6&style=for-the-badge&logoColor=white)](https://github.com/projectnessie/nessie/pkgs/container/nessie-gc)
```
Comment (snazy) [inline site/docs/downloads/index.md:101]: Better remove the badge for consistency

File: site/docs/downloads/index.md [hunk 7]
```diff
@@ -81,13 +98,13 @@ java -jar nessie-cli-{{ versions.nessie }}.jar
...
     ```bash
     docker pull ghcr.io/projectnessie/nessie-gc:{{ versions.nessie }}
     docker run ghcr.io/projectnessie/nessie-gc:{{ versions.nessie }} --help
     ```
 === "Quay.io"
-    [quay.io](https://quay.io/repository/projectnessie/nessie-gc?tab=tags)
+    [![quay.io Quay](https://img.shields.io/maven-central/v/org.projectnessie.nessie/nessie?label=quay.io+Docker&logo=docker&color=3f6ec6&style=for-the-badge&logoColor=white)](https://quay.io/repository/projectnessie/nessie-gc?tab=tags)
```
Comment (snazy) [inline site/docs/downloads/index.md:107]: Better remove the badge for consistency

File: site/docs/downloads/index.md [hunk 8]
```diff
@@ -113,13 +130,13 @@ Nessie repository.
 Docker images are multiplatform images for amd64, arm64, ppc64le, s390x.
 
 === "GitHub Container Registry"
-    [GitHub Container Registry](https://github.com/projectnessie/nessie/pkgs/container/nessie-server-admin)
+    [![ghcr.io GitHub Container Registry](https://img.shields.io/maven-central/v/org.projectnessie.nessie/nessie?label=ghcr.io+Docker&logo=docker&color=3f6ec6&style=for-the-badge&logoColor=white)](https://github.com/projectnessie/nessie/pkgs/container/nessie-server-admin)
```
Comment (snazy) [inline site/docs/downloads/index.md:133]: Better remove the badge for consistency

File: site/docs/downloads/index.md [hunk 9]
```diff
@@ -113,13 +130,13 @@ Nessie repository.
...
     ```bash
     docker pull ghcr.io/projectnessie/nessie-server-admin:{{ versions.nessie }}
     docker run ghcr.io/projectnessie/nessie-server-admin:{{ versions.nessie }} --help
     ```
 === "Quay.io"
-    [quay.io](https://quay.io/repository/projectnessie/nessie-server-admin?tab=tags)
+    [![quay.io Quay](https://img.shields.io/maven-central/v/org.projectnessie.nessie/nessie?label=quay.io+Docker&logo=docker&color=3f6ec6&style=for-the-badge&logoColor=white)](https://quay.io/repository/projectnessie/nessie-server-admin?tab=tags)
```
Comment (snazy) [inline site/docs/downloads/index.md:139]: Better remove the badge for consistency

File: site/in-dev/cli.md [hunk 10]
```diff
@@ -5,35 +5,55 @@ title: "Nessie CLI"
 # Nessie CLI
 
 The Nessie CLI is an easy way to get started with Nessie. It supports multiple branch 
-and tag management capabilities. This is installed as a standalone uber jar from the
```
Comment (snazy) [inline site/in-dev/cli.md:8]: Why remove the links to downloads and releases?

File: site/in-dev/cli.md [hunk 11]
```diff
@@ -5,35 +5,55 @@ title: "Nessie CLI"
...
 Nessie CLI is designed to be usable as an interactive REPL supporting auto-completion,
 highlighting where appropriate and has built-in help. Long outputs, like a commit log,
 are automatically paged like the Unix `less` command.
 
+![Nessie CLI](../img/cli-intro.png)
 
+## Installation
 
-![Nessie CLI](../img/cli-intro.png)
+Nessie CLI can be installed in two different ways:
```
Comment (snazy) [inline site/in-dev/cli.md:18]: Linking the downloads page is good enough. Can't reference the "right" version here (see my comment below)

File: site/in-dev/index-release.md [hunk 12]
```diff
@@ -26,13 +26,13 @@ The main Nessie server serves the Nessie repository using the Iceberg REST API a
 Docker images are multiplatform images for amd64, arm64, ppc64le, s390x.
 
 === "GitHub Container Registry"
-    [GitHub Container Registry](https://ghcr.io/projectnessie/nessie)
+    [![ghcr.io GitHub Container Registry](https://img.shields.io/maven-central/v/org.projectnessie.nessie/nessie?label=ghcr.io+Docker&logo=docker&color=3f6ec6&style=for-the-badge&logoColor=white)](https://github.com/projectnessie/nessie/pkgs/container/nessie)
```
Comment (snazy) [inline site/in-dev/index-release.md:29]: Better remove the badge for consistency

File: site/in-dev/index-release.md [hunk 13]
```diff
@@ -26,13 +26,13 @@ The main Nessie server serves the Nessie repository using the Iceberg REST API a
...
     ```bash
     docker pull ghcr.io/projectnessie/nessie:::NESSIE_VERSION::
     docker run -p 19120:19120 -p 9000:9000 ghcr.io/projectnessie/nessie:::NESSIE_VERSION::
     ```
 === "Quay.io"
-    [quay.io](https://quay.io/repository/projectnessie/nessie?tab=tags)
+    [![quay.io Quay](https://img.shields.io/maven-central/v/org.projectnessie.nessie/nessie?label=quay.io+Docker&logo=docker&color=3f6ec6&style=for-the-badge&logoColor=white)](https://quay.io/repository/projectnessie/nessie?tab=tags)
```
Comment (snazy) [inline site/in-dev/index-release.md:35]: Better remove the badge for consistency

File: site/in-dev/index-release.md [hunk 14]
```diff
@@ -61,6 +61,23 @@ java -jar nessie-quarkus-::NESSIE_VERSION::-runner.jar
 
 [Nessie CLI](cli.md) is both a command-line interface but primarily a REPL.
 
+### Docker image
+
+Docker images are multiplatform images for amd64, arm64, ppc64le, s390x.
+
+=== "GitHub Container Registry"
+    [![ghcr.io GitHub Container Registry](https://img.shields.io/maven-central/v/org.projectnessie.nessie/nessie?label=ghcr.io+Docker&logo=docker&color=3f6ec6&style=for-the-badge&logoColor=white)](https://github.com/projectnessie/nessie/pkgs/container/nessie-cli)
```
Comment (snazy) [inline site/in-dev/index-release.md:69]: Better remove the badge for consistency

File: site/in-dev/index-release.md [hunk 15]
```diff
@@ -61,6 +61,23 @@ java -jar nessie-quarkus-::NESSIE_VERSION::-runner.jar
...
+    ```bash
+    docker pull ghcr.io/projectnessie/nessie-cli:::NESSIE_VERSION::
+    docker run -it ghcr.io/projectnessie/nessie-cli:::NESSIE_VERSION:: 
+    ```
+=== "Quay.io"
+    [![quay.io Quay](https://img.shields.io/maven-central/v/org.projectnessie.nessie/nessie?label=quay.io+Docker&logo=docker&color=3f6ec6&style=for-the-badge&logoColor=white)](https://quay.io/repository/projectnessie/nessie-cli?tab=tags)
```
Comment (snazy) [inline site/in-dev/index-release.md:75]: Better remove the badge for consistency

File: site/in-dev/index-release.md [hunk 16]
```diff
@@ -80,13 +97,13 @@ java -jar nessie-cli-::NESSIE_VERSION::.jar
...
 === "GitHub Container Registry"
-    [ghcr.io GitHub Container Registry](https://github.com/projectnessie/nessie/pkgs/container/nessie-gc)
+    [![ghcr.io GitHub Container Registry](https://img.shields.io/maven-central/v/org.projectnessie.nessie/nessie?label=ghcr.io+Docker&logo=docker&color=3f6ec6&style=for-the-badge&logoColor=white)](https://github.com/projectnessie/nessie/pkgs/container/nessie-gc)
     ```bash
     docker pull ghcr.io/projectnessie/nessie-gc:::NESSIE_VERSION::
     docker run ghcr.io/projectnessie/nessie-gc:::NESSIE_VERSION:: --help
     ```
 === "Quay.io"
-    [quay.io Quay](https://quay.io/repository/projectnessie/nessie-gc?tab=tags)
+    [![quay.io Quay](https://img.shields.io/maven-central/v/org.projectnessie.nessie/nessie?label=quay.io+Docker&logo=docker&color=3f6ec6&style=for-the-badge&logoColor=white)](https://quay.io/repository/projectnessie/nessie-gc?tab=tags)
```
Comment (snazy) [inline site/in-dev/index-release.md:106]: Better remove the badge for consistency

File: site/in-dev/index-release.md [hunk 17]
```diff
@@ -112,13 +129,13 @@ Nessie repository.
 Docker images are multiplatform images for amd64, arm64, ppc64le, s390x.
 
 === "GitHub Container Registry"
-    [ghcr.io GitHub Container Registry](https://github.com/projectnessie/nessie/pkgs/container/nessie-server-admin)
+    [![ghcr.io GitHub Container Registry](https://img.shields.io/maven-central/v/org.projectnessie.nessie/nessie?label=ghcr.io+Docker&logo=docker&color=3f6ec6&style=for-the-badge&logoColor=white)](https://github.com/projectnessie/nessie/pkgs/container/nessie-server-admin)
```
Comment (snazy) [inline site/in-dev/index-release.md:132]: Better remove the badge for consistency

File: site/in-dev/index-release.md [hunk 18]
```diff
@@ -112,13 +129,13 @@ Nessie repository.
...
     ```bash
     docker pull ghcr.io/projectnessie/nessie-server-admin:::NESSIE_VERSION::
     docker run ghcr.io/projectnessie/nessie-server-admin:::NESSIE_VERSION:: --help
     ```
 === "Quay.io"
-    [quay.io Quay](https://quay.io/repository/projectnessie/nessie-server-admin?tab=tags)
+    [![quay.io Quay](https://img.shields.io/maven-central/v/org.projectnessie.nessie/nessie?label=quay.io+Docker&logo=docker&color=3f6ec6&style=for-the-badge&logoColor=white)](https://quay.io/repository/projectnessie/nessie-server-admin?tab=tags)
```
Comment (snazy) [inline site/in-dev/index-release.md:138]: Better remove the badge for consistency

File: site/in-dev/index.md [hunk 19]
```diff
@@ -21,18 +21,39 @@ Docker images are multiplatform images for amd64, arm64, ppc64le, s390x.
 **The image tags are updated twice per day during weekdays.**
 
 === "GitHub Container Registry"
-    [GitHub Container Registry](https://ghcr.io/projectnessie/nessie-unstable)
+    [![ghcr.io GitHub Container Registry](https://img.shields.io/maven-central/v/org.projectnessie.nessie/nessie?label=ghcr.io+Docker&logo=docker&color=3f6ec6&style=for-the-badge&logoColor=white)](https://github.com/projectnessie/nessie/pkgs/container/nessie-unstable)
```
Comment (snazy) [inline site/in-dev/index.md:24]: Better remove the badge for consistency

File: site/in-dev/index.md [hunk 20]
```diff
@@ -21,18 +21,39 @@ Docker images are multiplatform images for amd64, arm64, ppc64le, s390x.
...
     ```bash
     docker pull ghcr.io/projectnessie/nessie-unstable
     docker run -p 19120:19120 -p 9000:9000 ghcr.io/projectnessie/nessie-unstable
     ```
 === "Quay.io"
-    [quay.io](https://quay.io/repository/projectnessie/nessie-unstable?tab=tags)
+    [![quay.io Quay](https://img.shields.io/maven-central/v/org.projectnessie.nessie/nessie?label=quay.io+Docker&logo=docker&color=3f6ec6&style=for-the-badge&logoColor=white)](https://quay.io/repository/projectnessie/nessie-unstable?tab=tags)
```
Comment (snazy) [inline site/in-dev/index.md:30]: Better remove the badge for consistency

File: site/in-dev/index.md [hunk 21]
```diff
@@ -21,18 +21,39 @@ Docker images are multiplatform images for amd64, arm64, ppc64le, s390x.
...
+**The image tags are updated twice per day during weekdays.**
+
+=== "GitHub Container Registry"
+    [![ghcr.io GitHub Container Registry](https://img.shields.io/maven-central/v/org.projectnessie.nessie/nessie?label=ghcr.io+Docker&logo=docker&color=3f6ec6&style=for-the-badge&logoColor=white)](https://github.com/projectnessie/nessie/pkgs/container/nessie-cli-unstable)
+    ```bash
+    docker pull ghcr.io/projectnessie/nessie-cli-unstable
+    docker run -it ghcr.io/projectnessie/nessie-cli-unstable
+    ```
+=== "Quay.io"
+    [![quay.io Quay](https://img.shields.io/maven-central/v/org.projectnessie.nessie/nessie?label=quay.io+Docker&logo=docker&color=3f6ec6&style=for-the-badge&logoColor=white)](https://quay.io/repository/projectnessie/nessie-cli-unstable?tab=tags)
```
Comment (snazy) [inline site/in-dev/index.md:51]: Better remove the badge for consistency

File: site/in-dev/index.md [hunk 22]
```diff
@@ -45,13 +66,13 @@ Docker images are multiplatform images for amd64, arm64, ppc64le, s390x.
 **The image tags are updated twice per day during weekdays.**
 
 === "GitHub Container Registry"
-    [GitHub Container Registry](https://ghcr.io/projectnessie/nessie-gc-unstable)
+    [![ghcr.io GitHub Container Registry](https://img.shields.io/maven-central/v/org.projectnessie.nessie/nessie?label=ghcr.io+Docker&logo=docker&color=3f6ec6&style=for-the-badge&logoColor=white)](https://github.com/projectnessie/nessie/pkgs/container/nessie-gc-unstable)
```
Comment (snazy) [inline site/in-dev/index.md:69]: Better remove the badge for consistency

File: site/in-dev/index.md [hunk 23]
```diff
@@ -45,13 +66,13 @@ Docker images are multiplatform images for amd64, arm64, ppc64le, s390x.
...
     ```bash
     docker pull ghcr.io/projectnessie/nessie-gc-unstable
     docker run ghcr.io/projectnessie/nessie-gc-unstable --help
     ```
 === "Quay.io"
-    [quay.io](https://quay.io/repository/projectnessie/nessie-gc-unstable?tab=tags)
+    [![quay.io Quay](https://img.shields.io/maven-central/v/org.projectnessie.nessie/nessie?label=quay.io+Docker&logo=docker&color=3f6ec6&style=for-the-badge&logoColor=white)](https://quay.io/repository/projectnessie/nessie-gc-unstable?tab=tags)
```
Comment (snazy) [inline site/in-dev/index.md:75]: Better remove the badge for consistency

File: site/in-dev/index.md [hunk 24]
```diff
@@ -69,13 +90,13 @@ Docker images are multiplatform images for amd64, arm64, ppc64le, s390x.
 **The image tags are updated twice per day during weekdays.**
 
 === "GitHub Container Registry"
-    [GitHub Container Registry](https://ghcr.io/projectnessie/nessie-server-admin-unstable)
+    [![ghcr.io GitHub Container Registry](https://img.shields.io/maven-central/v/org.projectnessie.nessie/nessie?label=ghcr.io+Docker&logo=docker&color=3f6ec6&style=for-the-badge&logoColor=white)](https://github.com/projectnessie/nessie/pkgs/container/nessie-server-admin-unstable)
```
Comment (snazy) [inline site/in-dev/index.md:93]: Better remove the badge for consistency

File: site/in-dev/index.md [hunk 25]
```diff
@@ -69,13 +90,13 @@ Docker images are multiplatform images for amd64, arm64, ppc64le, s390x.
...
     ```bash
     docker pull ghcr.io/projectnessie/nessie-server-admin-unstable
     docker run ghcr.io/projectnessie/nessie-server-admin-unstable --help
     ```
 === "Quay.io"
-    [quay.io](https://quay.io/repository/projectnessie/nessie-server-admin-unstable?tab=tags)
+    [![quay.io Quay](https://img.shields.io/maven-central/v/org.projectnessie.nessie/nessie?label=quay.io+Docker&logo=docker&color=3f6ec6&style=for-the-badge&logoColor=white)](https://quay.io/repository/projectnessie/nessie-server-admin-unstable?tab=tags)
```
Comment (snazy) [inline site/in-dev/index.md:99]: Better remove the badge for consistency

File: site/docs/downloads/index.md (hunk 2 above)
Comment (adutra) [reply site/docs/downloads/index.md:30]: Why? I precisely added the badge wherever it was missing.

Comment (snazy) [reply site/docs/downloads/index.md:30]: See #8934 from yesterday

**Review by snazy — COMMENTED** [review]
The change looks overall good to me.
I think it needs to be adopted to #8939.

I'd like to merge this one shortly before a release - not immediately, because I fear some issues with the container registries again, but foremost because this changes the website telling about a download option that doesn't yet exist.

**Comment by adutra** [comment]
> I think it needs to be adopted to https://github.com/projectnessie/nessie/pull/8939.

Rebased, but I don't think we need to adapt anything: the `cli.md` page is not using any of the new tags introduced in 8939.

**Review by snazy — CHANGES_REQUESTED** [review]
+1 on this - the "Request Changes" is just to not accidentally merge before the next release - it's not a -1.

**Review by snazy — APPROVED** [review]

**Pull request merged**